csrc/core/scalar_type.hpp

```diff
@@ -382,6 +384,29 @@ class ScalarTypeTorch : public torch::CustomClassHolder, public ScalarType {
         exponent, mantissa, finite_values_only, NanRepr(nan_repr)));
   }
 
+  // This needs to be implemented and throw a TypeError in order for
+  // PyTorch's opcheck to work on ops that use ScalarTypes.
+  int64_t len() const {
+    throw c10::TypeError("__len__ not implemented");
```

Comment:
This looks like it will only compile with torch starting around 2.5+. Before that TypeError used to inherit c10::Error, which didn't provide such constructor

Reply:
Specifically, won't compile with torch before this PR: https://github.com/pytorch/pytorch/pull/122611/files#diff-0e933f7a8b52fdab64d0dca91edb58a6101a3eb07b55ca26224706005292fc7c

Reply:
This compiles with pytorch 2.4.0 which is the version listed in the requirements.

Reply:
It is only specified in the requirements for CUDA and CPU. Other platforms, such as neuron or openvino list earlier versions. On ROCm 6.2, there is not an official release > 2.3
Same type of compilation error applies to the entire class FakeScalarType: btw

Reply:
Ok, I can disable this on versions < 2.4.0.

Reply:
Thank you!